### ISSUE TITLE
Dashboard, Bookmark and PinnedApplication models should use a ForeignKey to User

### ISSUE THREAD

**Issue by rickydunlop** [geex-arts/django-jet#195]
I have a custom user model which swaps the id field with a UUID field. I tried setting up the dashboard and got a DataError

### To reproduce
* Create a custom user model with id as uuid field
```
import uuid
from django.contrib.auth.models import AbstractBaseUser, PermissionsMixin
class CustomUser(AbstractBaseUser, PermissionsMixin):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
```

* Set `AUTH_USER_MODEL` to this model
* Try to access the dashboard

You will see an error similar to this
`(1264, "Out of range value for column 'user' at row 1")`

### To fix
replace any user fields with
`user = models.ForeignKey(settings.AUTH_USER_MODEL, verbose_name=_('user'))` 
then change any references of `user.pk` to `user`

**Comment by SalahAdDin** [comment]
@f1nality What do you think about this?

**Comment by Ismael-VC** [comment]
Please come to the django-jet Discord server so we can organize if you like:

  *   https://discord.gg/3yr5yeH

Welcome! 😄

**Comment by gogson** [comment]
Same issue here.

**Comment by SalahAdDin** [comment]
Ok, @rickydunlop Can you make a PR in my Fork?

**Comment by rickydunlop** [comment]
@SalahAdDin Yeah sure can, what fork its yours? I don't see one on your Github page

**Comment by SalahAdDin** [comment]
Oh, it is a `jet_community` fork now.

**Comment by rickydunlop** [comment]
It looks like someone has already opened a PR with the patches I made 
https://github.com/jet-community/django-jet/pull/20

**Comment by SalahAdDin** [comment]
@rickydunlop Can you help us with reviewing? After it i will merge it.

**Comment by ramaprashanth** [comment]
@rickydunlop  @SalahAdDin It will be great if you guys can merge it. Let me know if you need any help.

**Comment by SalahAdDin** [comment]
We have not control here, only in the other repository.

**Comment by govindsharma7** [comment]
Note: We are maintaining a fork here at the moment: https://github.com/reckonsys/rs-django-jet addressing these issues. ONLY for UUID fields.

**Comment by SalahAdDin** [comment]
@govindsharma7 We also had a fork, are you working just in the UUID fields? No more features, right?

**Comment by govindsharma7** [comment]
@SalahAdDin Right now I have worked on UUID fields only.
